### PR TITLE
Refactor fourier to use split common code

### PR DESCRIPTION
**Pull request by zvecr**
## Description

I've followed what has been done previously to other boards (#3639).

A quick diff of the common split files and this keyboards copy seems like the differences are minimal and mostly in the area of backlight/rgb, but let me know if anything else is required.

## Types of changes

- [ ] Core
- [ ] Bugfix
- [ ] New Feature
- [x] Enhancement/Optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/Layout/Userspace (addition or update)
- [ ] Documentation

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document. (https://docs.qmk.fm/#/contributing)
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
@drashna Currently I've only managed to test a serial setup (as all my i2c other boards have some weird issues with left side only underglow when doing layer indication).

Happy to mark this as WIP till someone else can help test? However none of the current keymaps use I2C so i cannot tag anyone for help.

**Comment by drashna** [comment]
@nooges This look okay? 

Also, if you have an i2c board to test?

File: keyboards/fourier/rev1/config.h [hunk 1]
```diff
@@ -37,6 +34,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 // wiring of each half
 #define MATRIX_ROW_PINS { F4, D7, E6, B4 }
 #define MATRIX_COL_PINS { F5, F6, F7, B1, B3, B2, B6 }
+#define SPLIT_HAND_PIN D2
```
Comment (drashna) [inline keyboards/fourier/rev1/config.h]: Could you add this too:
```suggestion

/* serial.c configuration for split keyboard */
#define SOFT_SERIAL_PIN D0

```

Comment (zvecr) [reply keyboards/fourier/rev1/config.h]: The last comment on #4419 was about providing defaults for this value, given that other parts of the code base as well as split functionality provide defaults, should this really be required?

I'm happy to add if required, but it kinda feels dirty given how everything else works out of the box.

Comment (zvecr) [reply keyboards/fourier/rev1/config.h]: Cool, let me fire up my environment and make the required changes.

Comment (zvecr) [reply keyboards/fourier/rev1/config.h]: Done.

**Comment by zvecr** [comment]
And with the recent master changes, we should be good for testing again.

**Comment by nooges** [comment]
Confirmed to work with I2C on my Fourier

**Comment by zvecr** [comment]
Awesome, thanks @nooges

**Pull request merged**